kcidb/db/__init__.py

```diff
@@ -51,6 +50,49 @@ def get_drivers(cls):
     mux=MuxDriver,
 )
 
+class DBHelpAction(argparse.Action, DRIVER_TYPES):
```

Comment:
Hmm, this is not how one would pass arguments to an `argparse` action. This is how you would derive from a class. You have to find out how to do this right. Perhaps passing it to the constructor (`__init__`)? Try to figure it out, but if you're stuck reach out on the Slack channel. I or somebody else will help.

Also, we need to move this action to the corresponding `argparse` module, not keep it here.

Reply:
If I am passing as a parameter I should not need to import the DRIVER_TYPES. I am not finding the instance where this constructor was used so I could pass the argument. 

![ubuntu  Running  - Oracle VM VirtualBox 10-03-2023 11_03_52](https://user-images.githubusercontent.com/91889807/224232254-40620a29-abde-4774-9d31-dfebc970a450.png)

Reply:
Right, the `*Action` constructor is called by the original `argparse` module itself. So, we cannot pass a parameter this way.

Let's try another way: each `*Action` must have the `__call__` method, which is used for processing the option it's handling. One of its arguments is the "parser" it is used within. We can make the `kcidb.db.ArgumentParser` (to be `kcidb.db.argparse.ArgumentParser` after your change) accept a driver dictionary as one of its constructor arguments (in addition to already-accepted `database`) and store it in the instance.

Then we can make the `DBHelpAction.__call__()` method assert (using the `assert` statement) that the parser it's being given is an instance of that `ArgumentParser`, and get the driver list from it.

How about this?